src/main/java/core/basesyntax/FieldXY.java

```diff
@@ -13,7 +13,67 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        while (robot.getX() != toX || robot.getY() != toY) {
+            goToEnd(robot, toX, toY);
+        }
+
+    }
+
+    private Direction[] trueDirection(Robot robot, int toX, int toY) {
+        return new Direction[] {(toX > robot.getX()) ? Direction.RIGHT : Direction.LEFT,
+                                (toY > robot.getY()) ? Direction.UP : Direction.DOWN};
+    }
+
+    private void turnToNeededDirection(Robot robot, int toX, int toY) {
+        Direction[] neededDirection = trueDirection(robot, toX, toY);
+        switch (robot.getDirection()) {
+            case UP:
+                if (neededDirection[0].equals(Direction.RIGHT)) {
```

Comment:
```suggestion
                if (neededDirection[0] == Direction.RIGHT) {
```
it is allowed to compare enums with `==`